Mathlib/Data/Bitvec/Basic.lean

```diff
@@ -92,34 +90,41 @@ theorem ofNat_toNat {n : ℕ} (v : Bitvec n) : Bitvec.ofNat _ v.toNat = v := by
     subst n
     simp only [Bitvec.ofNat, Vector.toList_cons, Vector.toList_nil, List.reverse_cons,
       Vector.toList_append, List.foldr]
-    erw [add_lsb_div_two, to_bool_add_lsb_mod_two]
+    erw [addLsb_div_two, decide_addLsb_mod_two]
     congr
-    apply ys_ih
+    rename_i tail_ih
```

Comment:
`rename_i` is usually the wrong approach. Can't you introduce the name earlier instead?

Reply:
It was a hidden name. The hidden name was tail_ih

Reply:
The purpose of hidden names being inaccessible is to force you to pick a good name in the first place when it's introduced.

Using `rename_i` is cheating!

Reply:
I've made [a Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/rename_i.20during.20porting/near/329266787) here.

Reply:
replaced the induction with induction' and added names for the hypothesis. I really don't see how this is in any way better in terms of readability.